File: server/src/conversation/conversation.controller.ts
import { Controller, Get, Post, Delete, Param, Body, Query, Res, Header } from '@nestjs/common';
import { Response } from 'express';
import { ConversationService } from './conversation.service';

import { CurrentUser } from '../auth/user.decorator';
import type { User } from '../auth/auth.service';

@Controller('conversations')
export class ConversationController {
	constructor(
		private readonly service: ConversationService,
	) { }

	@Get()
	findAll(@CurrentUser() user: User) {
		return this.service.findAll(user.id);
	}

	@Post()
	create(@CurrentUser() user: User, @Body() body: { title?: string }) {
		return this.service.create(user.id, body.title);
	}

	@Delete(':id')
	remove(@CurrentUser() user: User, @Param('id') id: string) {
		return this.service.remove(id, user.id);
	}

	@Get(':id/messages')
	getMessages(@CurrentUser() user: User, @Param('id') id: string) {
		return this.service.getMessages(id, user.id);
	}


}

File: java-server/src/main/java/com/gistagent/conversation/ConversationController.java
package com.gistagent.conversation;

import java.util.List;
import java.util.Map;

import com.gistagent.auth.AuthenticatedUser;
import com.gistagent.conversation.dto.CreateConversationRequest;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/conversations")
public class ConversationController {

	private final ConversationService service;

	public ConversationController(ConversationService service) {
		this.service = service;
	}

	@GetMapping
	public List<Map<String, Object>> findAll(@AuthenticationPrincipal AuthenticatedUser user) {
		return service.findAll(user.id());
	}

	@PostMapping
	public Map<String, Object> create(@AuthenticationPrincipal AuthenticatedUser user,
									  @RequestBody(required = false) CreateConversationRequest body) {
		return service.create(user.id(), body == null ? null : body.title());
	}

	@DeleteMapping("/{id}")
	public void remove(@AuthenticationPrincipal AuthenticatedUser user, @PathVariable String id) {
		service.remove(id, user.id());
	}

	@GetMapping("/{id}/messages")
	public List<Map<String, Object>> getMessages(@AuthenticationPrincipal AuthenticatedUser user,
												 @PathVariable String id) {
		return service.getMessages(id, user.id());
	}
}
